File: frontend/src/entities/file.ts
import { EProcessingStatus } from '@/app/enums/file'
import { EColorVariants } from '@/app/enums/ui'

export type TProductLastFetch = 'fetch' | 'search' | 'init'

export interface IFile {
  id: string
  title: string
  originalName: string
  mimeType: string
  size: number
  processingStatus: EProcessingStatus
  scanStatus: string | null
  scanDetails: string | null
  metadataJson: Record<string, unknown> | null
  requiresAttention: boolean
  createdAt: string
  updatedAt: string
}

export interface IFileCriteria {
  offset: number
  limit: number
}

export interface IFileMap {
  id: string
  title: string
  originalName: string
  mimeType: string
  size: string
  processing: {
    variant: EColorVariants
    label: EProcessingStatus
  }
  scanStatus: string | null
  scanDetails: string | null
  metadataJson: Record<string, unknown> | null
  requiresAttention: boolean
  createdAt: string
  updatedAt: string
}

export interface IFileRec {
  title: string
  category: string
  rating?: number
  brand?: string
  price: number
  sku: string
  description?: string
}

export interface IFileState {
  criteria: IFileCriteria
}

/**
 * CONSTANTS
 **/

export const FILE_REC: IFileRec = {
  title: '',
  category: '',
  price: 0,
  sku: '',
}

export const FILE_STATE: IFileState = {
  criteria: { offset: 0, limit: 10 },
}

export const FILE_HEADERS: IHeader[] = [
  { key: 'title', title: 'Название', classes: '' },
  { key: 'originalName', title: 'Файл', classes: '' },
  { key: 'mimeType', title: 'MIME', classes: '' },
  { key: 'size', title: 'Размер', classes: '' },
  { key: 'processingStatus', title: 'Статус', classes: '' },
  { key: 'scanStatus', title: 'Проверка', classes: '' },
  { key: 'createdAt', title: 'Создан', classes: '' },
  { key: 'actions', title: '', classes: '' },
]

File: backend/src/schemas/file.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field


class FileSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    originalName: str
    storedName: str
    mimeType: str
    size: int
    processingStatus: str
    scanStatus: str | None
    scanDetails: str | None
    metadataJson: dict | None
    requiresAttention: bool
    createdAt: datetime
    updatedAt: datetime


class FileListCriteriaSchema(BaseModel):
    limit: int = Field(default=20, ge=1, le=100)
    offset: int = Field(default=0, ge=0)


class FileUpdateSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    title: str
